nikola/plugins/compile/ipynb.py

```diff
@@ -85,35 +84,123 @@ def read_metadata(self, post, file_metadata_regexp=None, unslugify_titles=False,
     def create_post(self, path, **kw):
         content = kw.pop('content', None)
         onefile = kw.pop('onefile', False)
+        kernel = kw.pop('ipython_kernel', None)
         # is_page is not needed to create the file
         kw.pop('is_page', False)
 
+        metadata = {}
+        metadata.update(self.default_metadata)
+        metadata.update(kw)
+
         makedirs(os.path.dirname(path))
+
+        if IPython.version_info[0] >= 3:
+            nb = nbformat.v4.new_notebook()
+            nb["cells"] = [nbformat.v4.new_code_cell(content)]
+        else:
+            nb = nbformat.v3.nbbase.new_notebook()
+            nb["cells"] = [nbformat.v3.nbbase.new_code_cell(content)]
+
         if onefile:
-            raise Exception('The one-file format is not supported by this compiler.')
+            nb["metadata"]["nikola"] = metadata
+
+        if kernel is not None:
+            nb["metadata"]["kernelspec"] = ipython_kernel_spec[kernel]
+            nb["metadata"]["language_info"] = ipython_language_info[kernel]
+
         with io.open(path, "w+", encoding="utf8") as fd:
-            if not content.startswith("Write your"):
-                fd.write(content)
+            if IPython.version_info[0] >= 3:
+                nbformat.write(nb, fd, 4)
             else:
-                fd.write("""{
- "metadata": {
-  "name": ""
- },
- "nbformat": 3,
- "nbformat_minor": 0,
- "worksheets": [
-  {
-   "cells": [
-    {
-     "cell_type": "code",
-     "collapsed": false,
-     "input": [],
-     "language": "python",
-     "metadata": {},
-     "outputs": []
-    }
-   ],
-   "metadata": {}
-  }
- ]
-}""")
+                nbformat.write(nb, fd, 'ipynb')
+
+# python2 nb metadata info
```

Comment:
>I would recommend just writing it as one large dict of dicts 

OK

>can’t we ask ipython for this information? 

No, because we are want to start with the ipynb with the correct kernel... and IPython does not know which kernel start until it gets the info from the ipynb (or goes to the default kernel if there is no info available....

>It really sucks if we have to keep it in our codebase and we have to keep it up-to-date…

Yep, sucks... but I am only adding the main kernels, which I think will be pretty stable so far...
I probably can make so more research to cut all this info to the minimal set of specification to start the correct kernel... then IPython fills the holes... I will research on that...